Assets/Prefabs/Player/PlayerController.cs

```diff
@@ -0,0 +1,173 @@
+using UnityEditor;
+using UnityEngine;
+
+public class PlayerController : MonoBehaviour
+{
+
+    #region UNITY_LIFECYCLE
+    // called when script instance is first loaded
+    private void Awake()
+    {
+        OnAwakeGetCharacterController();
+        OnAwakeBindInputs();
+    }
+
+    // called when GameObject is enabled
+    private void OnEnable()
+    {
+        inputActions.gameplay.Enable();
+    }
+
+    // called when GameObject is disabled
+    private void OnDisable()
+    {
+        inputActions.gameplay.Disable();
+    }
+    
+    // called on every Physics update
+    private void FixedUpdate()
+    {
+        ApplyMotion();
+    }
+    #endregion
+
+    #region PLAYER_INPUT
+    private InputActions inputActions;
+    private Vector2 moveInput = Vector2.zero;
+    private float moveAngle = 0.0f;
+    private float moveMagnitude = 0.0f;
+
+    /// <summary>
+    /// Instantiate the <c>InputActions</c> class and bind callbacks to store the 
+    /// current state of player input on changes.
+    /// </summary>
+    /// /// <remarks>Must be called during <c>OnAwake()</c></remarks>
+    private void OnAwakeBindInputs()
+    {
+        inputActions = new InputActions();
+        inputActions.gameplay.move.performed += context =>
+        {
+            moveInput = context.ReadValue<Vector2>();
+            moveAngle = -Vector2.SignedAngle(Vector2.up, moveInput);
+            moveMagnitude = moveInput.magnitude;
+        };
+        inputActions.gameplay.move.canceled += context =>
+        {
+            moveInput = Vector2.zero;
+            moveAngle = 0.0f;
+            moveMagnitude = 0.0f;
+        };
+    }
+
+    #endregion
+
+    #region PLAYER_MOVEMENT
+    [Header("Player Movement")]
+
+    [Tooltip("Camera to perform player motion relative to")]
+    public Camera myCamera;
+    [Tooltip("Walking speed of character (units / sec)")]
+    public float walkSpeed = 5.0f;
+    [Tooltip("Amount of time it takes to reach full walk / run speed (sec)")]
+    public float speedUpTime = 0.2f;
+    [Tooltip("Time it takes for the character to stop when the joystick is released (sec)")]
+    public float stopTime = 0.1f;
+    // TODO: IMPLEMENT THIS IN TERMS OF TIME BY SCALING WITH TIME.DELTATIME
```

Comment:
Isn't it already in units of time? 0.1 units (seconds). We could scale it by velocity. Or change how the controller uses this, stopTime, variable. 

Depending on the surface of the ground this could change too. I increased teh speed and stop time in Debug mode and it was like I was ice skating around.

Reply:
this variable is used in a `Quaternion.Slerp()` method, where we are linearly interpolating between "the direction the character is _currently_ facing" and "the direction the player _wants_ to face". 

So with this value of `0.1`, we are getting 10% closer to the desired direction on every Physics Update (~50 fps). This makes it hard to know what the ratio of `0.1` "means" in "how fast do I turn" since it's _really_ an asymptotic thing under the hood.

For right now, I thought the turning felt good - but I left a note to come back to this and figure out how to scale it appropriately with time because it's a wee bit wonkier than the normal speed-based logic it